service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentQueries.kt

```diff
@@ -65,26 +56,36 @@ fun Database.Transaction.insertAttachment(
 fun Database.Read.getAttachment(id: AttachmentId): Attachment? = this
     .createQuery(
         """
-        SELECT id, name, content_type
+        SELECT id, name, content_type, uploaded_by_employee, uploaded_by_person, application_id, income_statement_id
         FROM attachment
         WHERE id = :id
         """
     )
     .bind("id", id)
-    .mapTo<Attachment>()
+    .map { row ->
+        val applicationId = row.mapColumn<ApplicationId?>("application_id")
+        val incomeStatementId = row.mapColumn<IncomeStatementId?>("income_statement_id")
+        val attachedTo =
+            if (applicationId != null) AttachmentParent.Application(applicationId)
+            else if (incomeStatementId != null) AttachmentParent.IncomeStatement(incomeStatementId)
+            else AttachmentParent.None
```

Comment:
Hmmm.. `attachedTo` could be a generated column, see `message_account_type`. Or if the value is not needed in db queries, could it be a calculated value in `Attachment` data class?

Reply:
We talked about this through another channel and came to the conclusion that this is the best way to do it 😄